src/CommunityToolkit.Aspire.Hosting.SurrealDb/SurrealDbBuilderExtensions.cs

```diff
@@ -0,0 +1,418 @@
+        return builder.ApplicationBuilder.AddResource(surrealServerNamespace);
+    }
+
+    /// <summary>
+    /// Adds a SurrealDB database to the application model. This is a child resource of a <see cref="SurrealDbNamespaceResource"/>.
+    /// </summary>
+    /// <param name="builder">The SurrealDB resource builders.</param>
+    /// <param name="name">The name of the resource. This name will be used as the connection string name when referenced in a dependency.</param>
+    /// <param name="databaseName">The name of the database. If not provided, this defaults to the same value as <paramref name="name"/>.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
+    /// <remarks>
+    /// <example>
+    /// Add a SurrealDB container to the application model and reference it in a .NET project.
+    /// <code lang="csharp">
+    /// var builder = DistributedApplication.CreateBuilder(args);
+    ///
+    /// var db = builder.AddSurrealServer("surreal")
+    ///   .AddNamespace("ns")
+    ///   .AddDatabase("db");
+    /// var api = builder.AddProject&lt;Projects.Api&gt;("api")
+    ///   .WithReference(db);
+    ///  
+    /// builder.Build().Run(); 
+    /// </code>
+    /// </example>
+    /// </remarks>
+    public static IResourceBuilder<SurrealDbDatabaseResource> AddDatabase(
+        this IResourceBuilder<SurrealDbNamespaceResource> builder,
+        [ResourceName] string name,
+        string? databaseName = null
+    )
+    {
+        ArgumentNullException.ThrowIfNull(builder);
+        ArgumentException.ThrowIfNullOrEmpty(name);
+
+        // Use the resource name as the database name if it's not provided
+        databaseName ??= name;
+
+        builder.Resource.AddDatabase(name, databaseName);
+        var surrealServerDatabase = new SurrealDbDatabaseResource(name, databaseName, builder.Resource);
+
+        SurrealDbClient? surrealDbClient = null;
+
+        builder.ApplicationBuilder.Eventing.Subscribe<ConnectionStringAvailableEvent>(surrealServerDatabase, async (@event, ct) =>
+        {
+            var connectionString = await surrealServerDatabase.ConnectionStringExpression.GetValueAsync(ct).ConfigureAwait(false);
+            if (connectionString is null)
+            {
+                throw new DistributedApplicationException($"ConnectionStringAvailableEvent was published for the '{surrealServerDatabase}' resource but the connection string was null.");
+            }
+
+            var options = new SurrealDbOptionsBuilder().FromConnectionString(connectionString).Build();
+            surrealDbClient = new SurrealDbClient(options);
+        });
+
+        string namespaceName = builder.Resource.Name;
+        string serverName = builder.Resource.Parent.Name;
+
+        string healthCheckKey = $"{serverName}_{namespaceName}_{name}_check";
+        // TODO : Bug to be fixed
+        //builder.ApplicationBuilder.Services.AddHealthChecks().AddSurreal(_ => surrealDbClient!, healthCheckKey);
```

Comment:
What's the bug to be fixed?

Reply:
I don't think it's necessary. Just waiting for the release with .NET 10 I suppose.